tests/unit/commands/new-test.js

```diff
@@ -106,4 +106,91 @@ describe('new command', function () {
+  });
+
+  // Misuse: possibly an attempt to set app programming language
+  it('ember new with --lang flag and programming language fails with an error message', async function () {
+    let err = await expect(
+      command.validateAndRun(['foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang=typescript'])
+    ).to.be.rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.be.ok;
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.include('set the app programming language');
+    expect(err.message).to.include('typescript');
+    expect(err.message).to.not.include('ember-cli command option');
+  });
+
+  // Misuse: No specification + declared option
+  it('ember new with --lang flag but no specification fails with an error message; absorbs ember-cli option (declared)', async function () {
+    let err = await expect(command.validateAndRun(['foo', '--skip-npm', '--skip-bower', '--lang', '--skip-git'])).to.be
+      .rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.be.ok;
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.not.include('set the app programming language');
+    expect(err.message).to.include('ember-cli command option');
+    expect(err.message).to.include('--skip-git');
+  });
+
+  // // Misuse: No specification + hidden option
+  // it('ember new with --lang flag but no specification fails with an error message; absorbs appended ember-cli option (hidden)', async function () {
+  //   let err = await expect(command.validateAndRun(['foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang'])).to.be.rejected;
+  //   expect(err.name).to.equal('SilentError');
+  //   expect(err.message).to.be.ok;
+  //   expect(err.message).to.not.include('set the app programming language');
+  //   expect(err.message).to.include('ember-cli command option');
+  //   expect(err.message).to.include('--disable-analytics');
+  // });
+
+  // Misuse: Invalid Country Code
+  it('ember new with --lang flag and invalid code fails with an error message', async function () {
+    let err = await expect(command.validateAndRun(['foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang=en-UK']))
+      .to.be.rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.be.ok;
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.not.include('set the app programming language');
+    expect(err.message).to.not.include('ember-cli command option');
+  });
```

Comment:
I would prefer for this logic to live inside `init` command, and only be tested in one place (likely `tests/acceptance/new-test.js`).

Reply:
This should be covered by f95d3a9 and 72e10e9, respectively 👍